protocol/app/app.go

```diff
@@ -498,16 +519,24 @@ func New(
 	// See: https://github.com/cosmos/cosmos-sdk/blob/release/v0.46.x/x/gov/spec/01_concepts.md#proposal-messages
 	govRouter := govv1beta1.NewRouter()
 	govRouter.AddRoute(govtypes.RouterKey, govv1beta1.ProposalHandler).
-		AddRoute(paramproposal.RouterKey, params.NewParamChangeProposalHandler(app.ParamsKeeper)).
-		AddRoute(upgradetypes.RouterKey, upgrade.NewSoftwareUpgradeProposalHandler(app.UpgradeKeeper))
+		AddRoute(paramproposal.RouterKey, params.NewParamChangeProposalHandler(app.ParamsKeeper))
+	/** TODO(CORE-538): Migrate software upgrade type to Cosmos 0.50.
```

Comment:
It is because these types don't exist anymore and I haven't had time to integrate the new way to do upgrades yet.

There are a few TODOs in the codebase of this nature.

Reply:
Would merging this PR make the `main` branch to be in a broken state? For example, if the route for the upgrade is not registered, then upgrade via gov wouldn't work.

If there are no easy ways to resolve this in this PR, then I am okay. Just wanted to call out that we wanted to keep `main` always deployable and releasable.

Reply:
Luke replied here: https://github.com/dydxprotocol/v4-chain/pull/867#issuecomment-1877416949